src/main/java/org/logstash/dissect/fields/AbstractField.java

```diff
@@ -0,0 +1,69 @@
+package org.logstash.dissect.fields;
+
+import com.logstash.Event;
+import org.logstash.dissect.Delimiter;
+import org.logstash.dissect.ValueResolver;
+
+import java.util.Map;
+
+public abstract class AbstractField implements Field {
```

Comment:
For the casual reader, Daniel and I decided to use a IdentityHashMap for the fields to ValueRef map store - this is because it really is an identity lookup.